pkg/autoscaler/collector.go

```diff
@@ -280,20 +279,31 @@ func (c *collection) record(stat Stat) {
 func (c *collection) stableAndPanicConcurrency(now time.Time) (float64, float64, error) {
 	spec := c.currentMetric().Spec
 
-	c.buckets.RemoveOlderThan(now.Add(-spec.StableWindow))
+	panicAverage := aggregation.Average{}
+	var stableAverageVal float64
+	// if StableWindow is zero, then just grab the latest reading timestamp for stableAverage
+	if spec.StableWindow == 0 {
+		stableAverageVal = c.buckets.GetLatestReading(now)
+
+		c.buckets.ForEachBucket(
```

Comment:
This feels incorrect. If stable is only latest, then panic should be the same, most likely.